File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int left =0, right= nums.length-1;
        if(nums.length==0 || nums[left]<= nums[right]) return nums[left];
        while(left<right){
            int mid = left + (right-left)/2;
            if(nums[mid]>nums[mid+1]){
                return nums[mid+1];
            }else if(nums[mid]<nums[mid-1]){
                return nums[mid];
            }else if(nums[mid]>nums[left]){
                left=mid+1;
            }else{
                right=mid-1;
            }
        }
        
        return -1;
        
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low=0, high=nums.size()-1;
        int mini=INT_MAX;
        while(low<=high){
            if (nums[low] < nums[high]) {
                mini = min(mini, nums[low]);
                break;
            }
            int mid=low+(high-low)/2;
            mini=min(mini, nums[mid]);
            if(nums[mid]>=nums[low]){
                low=mid+1;
            }else{
                high=mid-1;
            }
        }
        return mini;
    }
};